app/assets/javascripts/tasks_index.coffee

```diff
@@ -56,36 +56,40 @@ addAnimatedSliding =->
 
 initializeFilter =->
   $("#reset-btn").click ->
-    $('#search').val('')
-    $(".index").submit()
+    $('.ransack-filter').val('').trigger('change')
+    $("#task_search").submit()
 
 
   $(".change-hidden-field").click ->
     $('#option').val(this.id)
-    $(".index").submit()
+    $('#task_search').submit()
   $('#'+$('#option').val()).addClass('selected')
   intializeAdvancedFilter()
 
 intializeAdvancedFilter =->
-  if $('#dropdownWindowActive')
-    if $('#dropdownWindowActive').val() == "true"
-      $('.dropdown-content').show()
-
-  $('#advanced').click ->
-    $('.dropdown-content').toggle()
-    search = $('#search')
-    if $('#dropdownWindowActive').val() == "true"
-      $(search).css("border-bottom-left-radius", "4px")
-      $("#advanced").css("border-bottom-right-radius", "4px")
-      $('#dropdownWindowActive').val(false)
-      $('#drop').removeClass('fa-caret-up').addClass('fa-caret-down')
+  $dropdownWindowActive = $('#dropdownWindowActive')
+  $dropdownContent = $('.dropdown-content')
+  if $dropdownWindowActive
+    if $dropdownWindowActive.val() == "true"
```

Comment:
You only moved that, I know. But shouldn't we use a safe comparison operator `===` here (and a few lines below)?

Reply:
CoffeeScript already seems to handle that. See [1]

> Because the == operator frequently causes undesirable coercion, is intransitive, and has a different meaning than in other languages, CoffeeScript compiles == into ===, and != into !==. In addition, is compiles into ===, and isnt into !==.

[1]: https://coffeescript.org/#expressions

Reply:
Oh, thanks for pointing that out. I wasn’t writing CoffeeScript since ages (but rather plain JS) and didn’t know that. Thanks!